packages/service-library/tests/fastapi/test_docker_utils.py

```diff
@@ -0,0 +1,120 @@
+# pylint: disable=protected-access
+from collections.abc import Awaitable, Callable
+from typing import Any
+from unittest.mock import call
+
+import pytest
+from models_library.docker import DockerGenericTag
+from pydantic import parse_obj_as
+from pytest_mock import MockerFixture
+from servicelib import progress_bar
+from servicelib.docker_utils import pull_image
+from servicelib.fastapi.docker_utils import retrieve_image_layer_information
+from settings_library.docker_registry import RegistrySettings
+
+
+@pytest.mark.parametrize(
+    "service_repo, service_tag",
+    [
+        ("itisfoundation/sleeper", "1.0.0"),
+        ("itisfoundation/sleeper", "2.2.0"),
+        (
+            "itisfoundation/sleeper",
+            "sha256:a6d9886311721d8d341068361ecf9998a3c7ecb0efb23ebac553602c2eca1f8f",
+        ),
+    ],
+)
+async def test_retrieve_image_layer_information(
+    remove_images_from_host: Callable[[list[str]], Awaitable[None]],
+    registry_settings: RegistrySettings,
+    osparc_service: dict[str, Any],
+    service_repo: str,
+    service_tag: str,
+):
+    # clean first
+    image_name = f"{service_repo}:{service_tag}"
```

Comment:
THOUGHT: you have used this resolution of the image_name in several places. If you are using it in the code perhaps it would be a good idea to have an util that composes image names from parts? Similar to types like HttpUrl from pydantic

Reply:
not sure I understand this comment